### PR TITLE
Admin App Forms: The Good Parts

### PR DESCRIPTION
**Pull request by samchrisinger**
# Purpose
There are some important changes in samchrisinger:feature/admin-app-forms that should be merged into the release branch. This PR aggregates those changes. 

# Changes
- let $osf.block apply to a specific DOM element
- add $osf.dialog
- change flags to snake case in Prereg Challenge schema
- don't delete DraftReg approval reference upon rejection
- let DraftReg flags get updated (but not overwritten) if the schema
- changes
- add setter for DraftReg flags
- add is_rejected property to DraftReg
- update DraftRegApproval to use a URL template rather than url_for

### CODE REVIEW

File: website/static/js/osfHelpers.js [hunk 1]
```diff
@@ -744,6 +752,50 @@ var any = function(listOfBools) {
...
+ * @param {String} actionButtonLabel:
+ * @param {Object} options: optional options
+ * @param {String} options.actionButtonClass: CSS class for action button, default 'btn-success'
+ * @param {String} options.cancelButtonLabel: label for cancel button, default 'Cancel'
+ * @param {String} options.cancelButtonClass: CSS class for cancel button, default 'btn-default'
+ *
+ * @example
+ * dialog('Hello', 'Just saying hello', 'Say hi').done(successCallback).fail(doNothing);
+ **/
+var dialog = function(title, message, actionButtonLabel, options) {
```
Comment (sloria) [inline website/static/js/osfHelpers.js]: :+1:

Comment (sloria) [inline website/static/js/osfHelpers.js]: This is nice, but is this used anywhere?

Comment (samchrisinger) [inline website/static/js/osfHelpers.js]: It is in the admin app. If need be I can remove it, but it'll just get added later.

Comment (sloria) [inline website/static/js/osfHelpers.js]: It's fine for now.

**Pull request merged**